cmd/oras/internal/display/handler.go

```diff
@@ -174,9 +174,40 @@ func NewManifestPushHandler(printer *output.Printer) metadata.ManifestPushHandle
 	return text.NewManifestPushHandler(printer)
 }
 
-// NewManifestIndexCreateHandler returns an index create handler.
-func NewManifestIndexCreateHandler(printer *output.Printer) metadata.ManifestIndexCreateHandler {
-	return text.NewManifestIndexCreateHandler(printer)
+// NewManifestIndexCreateHandler returns status, metadata and content handlers for index create command.
+func NewManifestIndexCreateHandler(outputPath string, printer *output.Printer, pretty bool) (
+	status.ManifestIndexCreateHandler,
+	metadata.ManifestIndexCreateHandler,
+	content.ManifestIndexCreateHandler) {
+	statusHandler := status.NewTextManifestIndexCreateHandler(printer)
+	metadataHandler := text.NewManifestIndexCreateHandler(printer)
+	contentHandler := content.NewManifestIndexCreateHandler(printer, pretty, outputPath)
+	switch outputPath {
+	case "":
+		contentHandler = content.NewDiscardHandler()
+	case "-":
+		statusHandler = status.NewDiscardHandler()
+		metadataHandler = metadata.NewDiscardHandler()
+	}
```

Comment:
nit: Having initial values then overwritten is logically correct. However, it may cause unnecessary resource allocation. You may refactor them and make it clearer.

Reply:
Reverted this part back to the original version, before resolving this [comment](https://github.com/oras-project/oras/pull/1509#discussion_r1776443538)